src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1278,6 +1278,31 @@ private static bool HasNamespaceName(NamespaceSymbol @namespace, string namespac
             return (name.Length == length) && (string.Compare(name, 0, namespaceName, offset, length, comparison) == 0);
         }
 
+        internal static bool IsSpanType(this TypeSymbol type)
+        {
+            if ((type as NamedTypeSymbol)?.Arity != 1)
+            {
+                // must be a generic type of arity '1'
+                return false;
+            }
+
+            if (type.Name != "Span" && type.Name != "ReadOnlySpan")
+            {
+                // must be called "Span" or "ReadOnlySpan"
+                return false;
+            }
+
+            var ns = type.ContainingSymbol as NamespaceSymbol;
+            if (ns?.Name != "System")
+            {
+                // must be in "System" namespace
+                return false;
+            }
+
+            // the "System" must be in the global namespace
+            return ns.ContainingNamespace.IsGlobalNamespace;
```

Comment:
>There are many other cases where we assume that ContainingNamespace is not null though

That assumption could be valid when we know what symbol we are dealing with, where did it come from. For example, it is valid for PE/Source/Retargeting symbol. 

---
In reply to: [120983984](https://github.com/dotnet/roslyn/pull/20058#discussion_r120983984) [](ancestors = 120983984,120980661,120762517)